### PR TITLE
Fork choice filter changes

### PR DESCRIPTION
**Pull request by adiasg**
**What type of PR is this?**
Feature

**What does this PR do? Why is it needed?**
Changes the fork choice filteration logic as needed by https://github.com/ethereum/consensus-specs/pull/3431

### CODE REVIEW

**Comment by potuz** [comment]
Discussing offline the changes

File: proto/ssz_proto_library.bzl [hunk 1]
```diff
@@ -47,7 +47,7 @@ minimal = {
     "extra_data.size": "32",
     "max_blobs_per_block.size": "6",
     "max_blob_commitments.size": "16",
-    "kzg_commitment_inclusion_proof_depth.size": "9",
+    "kzg_commitment_inclusion_proof_depth.size": "17",
```
Comment (prestonvanloon) [inline proto/ssz_proto_library.bzl:50]: Please explain this change. Was there an upstream spec change for this?

Comment (terencechain) [reply proto/ssz_proto_library.bzl:50]: I believe it's this one: https://github.com/ethereum/consensus-specs/pull/3255

Comment (prestonvanloon) [reply proto/ssz_proto_library.bzl:50]: This change fails test `//testing/spectest/minimal/deneb/ssz_static:go_default_test`

**Comment by potuz** [comment]
Blocking just in case cause I still don't understand a failing testcase

File: proto/ssz_proto_library.bzl (hunk 1 above)
Comment (potuz) [reply proto/ssz_proto_library.bzl:50]: the blob size will not change the proof size, the proof is for the KZG commitment, the block is smaller so it needs a smaller Merkle Tree, I'm pushing a revert of this and approving this PR

**Review by potuz — APPROVED** [review]

**Review by potuz — APPROVED** [review]

**Pull request merged**